### PR TITLE
Adapt to changes in Startup class and DbContext discovery

### PR DESCRIPTION
**Pull request by prafullbhosale**
- Change IConfigurationRoot -> IConfiguration in DbContextServices.
 - Fix test project setup to use new Startup and Program class content.

Address #464 

- Scaffolding now adds a new `DbContextFactory` for a new DbContext generated by scaffolding.

Address #467
cc @Eilon

### CODE REVIEW

File: test/Shared/MsBuildProjectStrings.cs [hunk 1]
```diff
@@ -246,16 +242,31 @@ public void Configure(IApplicationBuilder app, IHostingEnvironment env, ILoggerF
...
 {
     public class Program
     {
         public static void Main(string[] args)
         {
-            Console.WriteLine(""Hello"");
+            BuildWebHost(args).Run();
         }
+
+        public static IWebHost BuildWebHost(string[] args) => WebHost
```
Comment (Eilon) [inline test/Shared/MsBuildProjectStrings.cs]: We're updating the indentation on this to be a bit nicer:

```c#
        public static IWebHost BuildWebHost(string[] args) =>
            WebHost
                .CreateDefaultBuilder(args)
                .UseStartup<Startup>()
                .Build();
```

**Comment by prafullbhosale** [comment]
cc @bricelam for `DbContextFactory` related changes.

File: test/Shared/MsBuildProjectStrings.cs (hunk 1 above)
Comment (prafullbhosale) [inline test/Shared/MsBuildProjectStrings.cs]: Updated the test setup to create Program.cs with nicer indentation.

File: src/Microsoft.VisualStudio.Web.CodeGeneration.EntityFrameworkCore/Templates/DbContext/NewDbContextFactory.cshtml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    }
+}
+@{
+    var factoryClassName = @Model.DbContextTypeName+"Factory";
+}
+namespace @Model.DbContextNamespace
+{
+    public class @factoryClassName : IDbContextFactory<@Model.DbContextTypeName>
+    {
+        public @Model.DbContextTypeName Create(DbContextFactoryOptions options) =>
```
Comment (bricelam) [inline src/Microsoft.VisualStudio.Web.CodeGeneration.EntityFrameworkCore/Templates/DbContext/NewDbContextFactory.cshtml]: This should be `string[] args` and passed into `.BuildWebHost`. See dotnet/templating#705

Comment (prafullbhosale) [inline src/Microsoft.VisualStudio.Web.CodeGeneration.EntityFrameworkCore/Templates/DbContext/NewDbContextFactory.cshtml]: Updated.

**Comment by prafullbhosale** [comment]
a6414d24afba24309c099105bd35b3dca4a4f415

**Pull request closed without merging**